payments/views.py

```diff
@@ -15,18 +15,49 @@ class WorkshopList(ListView):
     context_object_name = 'upcoming_workshops'
 
 
-class WorkshopDetail(DetailView):
+class WorkshopDetail(FormMixin, DetailView):
     model = Workshop
+    form_class = OrderForm
     context_object_name = 'workshop'
     template_name = 'payments/detail.html'
 
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs['workshop'] = self.object
```

Comment:
No worries, thats a great question. Yes, its probably worth adding a note, but `get_form_kwargs` is overriding the builtin `get_form_kwargs` method from (`FormMixin`)[https://ccbv.co.uk/projects/Django/1.9/django.views.generic.edit/FormMixin/]. This method ensures that any extra kwargs we might need to pass into the form at instantiation make it in.